src/files/examples/fingerprint_reader_functions.py

```diff
@@ -0,0 +1,260 @@
+
+################################################################################
+# SCRIPT_DIR SETTING
+################################################################################
+
+################################################################################
+################################################################################
+# INCLUDES
+################################################################################
+################################################################################
+import sys
+import logging # http://www.onlamp.com/pub/a/python/2005/06/02/logging.html
+import time
+from pyfingerprint.pyfingerprint import PyFingerprint
+import argparse
+import hashlib
+
+################################################################################
+################################################################################
+# CONSTANTS
+################################################################################
+################################################################################
+
+################################################################################
+################################################################################
+# VARIABLES
+################################################################################ 
+################################################################################ 
+
+global f # Fingerprintreader
+
+################################################################################
+################################################################################
+# FUNCTIONS DEFINITION 
+################################################################################
+################################################################################
+def init():
+	global f
+	logging.basicConfig(level=logging.DEBUG, stream=sys.stdout, filemode="a+", format="%(asctime)-15s %(levelname)-8s %(message)s")
+	try:
+	    f = PyFingerprint('/dev/ttyUSB0', 57600, 0xFFFFFFFF, 0x00000000)
+
+	    if ( f.verifyPassword() == False ):
+		raise ValueError('The given fingerprint sensor password is wrong!')
+
+	except Exception as e:
+	    print('The fingerprint sensor could not be initialized!')
+	    print('Exception message: ' + str(e))
+	    exit(1)
+
+def display_system_parameters():
+	""" Display system parameters as logger info level"""
+	tuple1 = f.getSystemParameters()
+	logging.info("System parameters : status register (%s), system id(%s), storage capacity(%s), security level(%s), sensor address(%s), packet length(%s), baudrate(%s)" % (tuple1[0], tuple1[1], tuple1[2], tuple1[3], tuple1[4], tuple1[5], tuple1[6]))
+
+def delete():
+	"""Deletes a template after asking for its position """
+	try:
+		positionNumber = input('Please enter the template position you want to delete: ')
+		positionNumber = int(positionNumber)
+
+		if ( f.deleteTemplate(positionNumber + 1) == True ):
```

Comment:
Index is zero based. Do not increase index as it is confusing. If template 0 should be deleted 1 is deleted!